core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -1856,6 +1895,65 @@ private void receive(Transaction tx, StoredBlock block, BlockChain.NewBlockType
         saveNow();
     }
 
+    /**
+     * Finds if tx is NOT spending other txns which are in the specified confidence type
+     * @param tx
+     * @param confidenceType
+     * @return true if tx is NOT spending other txns which are in the specified confidence type. false otherwise
+     */
+    private boolean isNotSpendingTxnsInConfidenceType(Transaction tx, ConfidenceType confidenceType) {
+        boolean isNotSpendingTxnsInConfidenceType = true;
+        for (TransactionInput txInput : tx.getInputs()) {
+            Transaction connectedTx = this.getTransaction(txInput.getOutpoint().getHash());
+            if (connectedTx!=null && connectedTx.getConfidence().getConfidenceType().equals(confidenceType)) {
+                isNotSpendingTxnsInConfidenceType = false;
+            }
+        }        
+        return isNotSpendingTxnsInConfidenceType;
+    }
+
+    
+    /**
+     * Creates and returns a new List with the same txns as inputSet but txns are sorted by depencency.
+     * If tx B spends tx A, then tx A should before tx B on the returned List.
+     * Several invocations to this method with the same inputSet could result in lists with txns in different order, 
+     * as there is no guarantee on the order of the returned txns besides what was already stated.
+     * @param inputSet the txns to sort
+     * @return a List of txns where if tx B spends tx A, then tx A should before tx B.
+     */
+    List<Transaction> sortTxnsByDependency(Set<Transaction> inputSet) {
+        List<Transaction> result = new ArrayList<Transaction>(inputSet);
```

Comment:
Some JVMs can't do type sharpening e.g. on Android so it's a bit faster to write out ArrayList explicitly here. Not a big deal.